06-release-engineering-analytics.Rmd

```diff
@@ -321,27 +324,116 @@ iteratively and planned for four weekly milestones.
 
 ## Answers
 
-<!-- TODO -->
+In this section, we will give an answer to each of the research questions
+that we have presented in [Section 7.1.1](#research-questions-1).
 
-### RQ1: ...
+### RQ1: Modern Release Engineering Practices
 
-<!-- TODO -->
+This section will give an answer to the question:
+_How is modern release engineering done in practice?_
+
+@adams2016a and @karvonen2017a have described release engineering
+practices that are currently in use in the industry.
+@adams2016a focused on modern release engineering,
+while @karvonen2017a investigated agile release engineering,
+which is a subset of all modern release engineering.
+Both studies agreed that modern release engineering consists
+of the following components:
+
+- **Continuous Integration (CI).**
+
+    To quote @adams2016a: CI "refers to the activity of continuously polling the
+    [version control system] for new commits or merges,
+    checking these revisions out on dedicated build machines,
+    compiling them and running an initial set of tests to check for regressions."
+
+- **Continuous Deployment or Continuous Delivery (CD).**
+
+    These two terms describe the ability to deploy (or deliver)
+    a software product to the end user very frequently.
+    A term related to this is **Rapid Releases (RR)**.
```

Comment:
Given the fact that  **Rapid Releases (RR)** are so often studied, I think it deserves to be its own practice. Rapid release can be done without continuous deployment.

How about we introduce rapid releases as a practice, explain that lots of people in the industry have shifted towards shorter release cycles et cetera. We then introduce CD as a separate practice and explain that this is a very specific practice that enables rapid releases.

Reply:
Thanks for your suggestion, I've pushed an update :slightly_smiling_face: